recipe-server/src/main/kotlin/com/github/ptrteixeira/cookbook/ApplicationComponent.kt

```diff
@@ -0,0 +1,24 @@
+package com.github.ptrteixeira.cookbook
+
+import com.github.ptrteixeira.cookbook.auth.AuthModule
+import com.github.ptrteixeira.cookbook.base.BaseModule
+import com.github.ptrteixeira.cookbook.base.ObjectMapperModule
+import com.github.ptrteixeira.cookbook.core.User
+import com.github.ptrteixeira.cookbook.data.DataModule
+import com.github.ptrteixeira.cookbook.resources.ResourcesModule
+import com.tylerkindy.dropwizard.dagger.DropwizardInjectionModule
+import com.tylerkindy.dropwizard.dagger.DropwizardInjector
+import dagger.Component
+import io.dropwizard.auth.Authenticator
+
+@Component(modules = [
+    ObjectMapperModule::class,
+    BaseModule::class,
+    DataModule::class,
+    AuthModule::class,
+    DropwizardInjectionModule::class,
+    ResourcesModule::class
+])
+interface ApplicationComponent : DropwizardInjector<CookbookApplication> {
+    fun authenticator(): Authenticator<String, User>
```

Comment:
You don't need to expose this instance from the `@Component` unless you need to manually create the value somewhere. If it's being `@Inject`ed somewhere, it just needs to be bound in the graph.